docs/plugins/npe2_manifest_specification.md

```diff
@@ -0,0 +1,578 @@
+napari.manifest =
+    npe2-tester = npe2_tester:napari.yaml
+```
+
+The manifest file is specified relative to the submodule root path.
+So for the example it will be loaded from: `<path/to/npe2-tester>/napari.yaml`.
+
+### 2. Include the manifest in the package distribution
+
+The manifest file needs to be included as _[package data][pd]_ in
+distributable forms for the package. For example:
+
+```toml
+[metadata]
+...
+include_package_data=True
+
+[options.package_data]
+npe2_tester =
+    napari.yaml
+```
+
+## Manifest schema
+
+The plugin manifest file is a [json][], [yaml][], or [toml][] conforming to
+a schema. The `npe2` package includes a command-line tool that can be used
+to validate a schema:
+
+```
+pip install npe2
+npe2 validate my_plugin_manifest.yml
+```
+
+The manifest file's fields are described in detail below. The manifest file's
+overall structure can be viewed as a hierarchy:
+
+```
+<top-level properties>
+contributions:
+  commands:     <list of commands>
+  readers:      <list of readers>
+  writers:      <list of writers>
+  sample_data:  <list of sample data providers>
+  widgets:      <list of widget providers>
+  themes:       <list of themes>
+```
+
+Readers, writers, sample data providers and widget providers refer to callable
+python functions that a plugin defines. Each callable is identified with an
+entry in the list of `commands` via a unique id. For more see {ref}`Commands` below.
+
+```{note}
+Python package metadata (`setup.py` or `setup.cfg`) may be used to populate
+selected properties of the [PluginManifest][].
+```
+
+(top-level-props)=
+
+## Top-level properties
+
+### Required
+
+- **name** The name of the plugin. Example: `napari-svg`. Should be a
+  [PEP-8][]-compatible package name. If missing, this is populated from the
+  python package [name][setup-name].
+
+### Optional
+
+- **engine**: A SemVer compatible version string matching the versions of the
+  plugin engine that the extension is compatible with. Example: "0.1.0".
+- **display_name**: User-facing text to display as the name of this plugin.
+  Example: `napari SVG`. Must be 3-40 characters long, containing
+  printable word characters, and must not begin or end with an underscore,
+  white space, or non-word character.
+- **entry_point**: The module containing the `activate()` function. Example:
+  `foo.bar.baz`. This should be a fully qualified module string.
+
+(commands)=
+
+## Commands
+
+**command** is a python function associated with an id. The **id** is
+a unique identifier to which other contributions, like readers, can refer.
+
+### Required fields
+
+- **id** An identifier used to reference this command within this plugin.
+- **title** A description of the command. This might be used, for example,
+  when searching in a command palette. Examples: "Generate lily sample",
+  "Read tiff image", "Open gaussian blur widget".
+
+### Optional fields
+
+- **python_name** Fully qualified name to callable python object implementing
```

Comment:
We actually don't need this for sample data by URI right? So I would add text like:

"Most commands will require a python_name pointing to a callable python object to execute. Exceptions are the sample data uri command and any commands whose functions are dynamically specified inside the plugin's activate function"

Reply:
Sample-data-by-URI contributions don't end up referencing a command at all. The only reason this field is optional is to support dynamic binding.